cloudify/deployments.go

```diff
@@ -161,3 +163,23 @@ func (cl *Client) CreateDeployments(deploymentID string, depl DeploymentPost) (*
 
 	return &deployment, nil
 }
+
+func (cl *Client) GetDeploymentByInstance(instanceID string) (string, error) {
+	var params  = map[string]string{}
+	var deploymentID = ""
+
+	nodeInstances, err := cl.GetNodeInstances(params)
+	if err != nil {
+		return deploymentID, err
+	}
+
+	for _, nodeInstance := range nodeInstances.Items {
+		if nodeInstance.ID == instanceID {
+			deploymentID = nodeInstance.DeploymentID
+			log.Printf("Found deployment ID: %v", deploymentID)
+			return deploymentID, nil
+		}
+	}
+
+	return deploymentID, errors.New("unable to find node instance")
```

Comment:
Or if you like to provide exact instance that we by mistake provided in params - you can use fmt.Errorf("unable to find node instance %v", instanceID)